src/mdns/mdns.hpp

```diff
@@ -189,7 +189,8 @@ class Publisher : private NonCopyable
      *                          provided, this service resides on local host and it is the implementation
      *                          to provide specific host name. Otherwise, the caller MUST publish the host
      *                          with method PublishHost.
-     * @param[in] aName         The name of this service.
+     * @param[in] aName         The name of this service. If an empty string is provided, the service's name will be the
```

Comment:
Why this is useful and necessary?

Reply:
SRP Replication's [RFC draft](https://www.ietf.org/archive/id/draft-lemon-srp-replication-02.html#name-advertising-ones-own-replic) asks to use the hostname as the service instance name.